rosidl_typesupport_c/src/type_support_dispatch.hpp

```diff
@@ -63,25 +62,30 @@ get_typesupport_handle_function(
           fprintf(stderr, "Failed to find library '%s'\n", library_name);
           return nullptr;
         }
-        lib = new Poco::SharedLibrary(library_path);
+
+        try {
+          lib = new rcpputils::SharedLibrary(library_path.c_str());
+        } catch (const std::runtime_error & e) {
+          throw std::runtime_error("Could not load library " + std::string(e.what()));
```

Comment:
The exception message doesn't include the library path for which the error occurred and is therefore not really helpful for users. Arguably the lower level API in `rcpputils` / `rcutils` should probably already include the library path in the exception / error message.